src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -677,55 +696,124 @@ void PeakDetectorCLI::saveJson(string setName) {
 		#endif
 
 		jsonReports=new JSONReports(mavenParameters);
-		jsonReports->saveMzEICJson(mavenParameters->outputdir + setName + ".json",
+		string fileName = mavenParameters->outputdir + setName + ".json";
+		if (uploadToPolly_bool){
+			string fileName = setName;
```

Comment:
@shubhra-agrawal @Ky5595 fileName scope will end after the if statement. You should not initialize it again.